Doc/whatsnew/3.14.rst

```diff
@@ -233,6 +233,11 @@ For further information on how to build Python, see
 (Contributed by Ken Jin in :gh:`128718`, with ideas on how to implement this
 in CPython by Mark Shannon, Garret Gu, Haoran Xu, and Josh Haberman.)
 
+.. note::
+
+   This is not to be confused with tail call optimization of Python code.
+   Python functions do not currently have tail call optimization.
```

Comment:
Should we specify CPython? I'm not sure if other implementations have support for TCO or not, and it would likely be an implementation detail rather than part of the language reference if it was implemented.
```suggestion
.. note::
   This is not to be confused with `tail call optimization`__ of Python functions.
   Python functions do not currently have tail call optimization.

   __ https://en.wikipedia.org/wiki/Tail_call

(Contributed by Ken Jin in :gh:`128718`, with ideas on how to implement this
in CPython by Mark Shannon, Garret Gu, Haoran Xu, and Josh Haberman.)
```

Reply:
True. Good point!

Reply:
Oops, it was automerged. Could you please help me open a PR? So sorry!

Reply:
See #129812